core/commonMain/src/kotlinx/serialization/Serializers.kt

```diff
@@ -14,8 +14,24 @@ import kotlinx.serialization.builtins.TripleSerializer
 import kotlinx.serialization.internal.*
 import kotlinx.serialization.modules.*
 import kotlin.jvm.*
+import kotlin.native.concurrent.SharedImmutable
 import kotlin.reflect.*
 
+/**
+ * Cache class for non-parametrized and non-contextual serializers.
+ */
+@SharedImmutable
```

Comment:
`@SharedImmutable` for cache looks very counter-intuitive — until one finds that cache implementation on Native doesn't cache anything. Probably if we change this to `@ThreadLocal`, it wouldn't be a big waste, and we may want to make a simple cache for Native later.